google/cloud/spanner/internal/transaction_impl_test.cc

```diff
@@ -0,0 +1,202 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/internal/transaction_impl.h"
+#include "google/cloud/spanner/internal/time.h"
+#include "google/cloud/spanner/timestamp.h"
+#include "google/cloud/spanner/transaction.h"
+#include <gmock/gmock.h>
+#include <chrono>
+#include <ctime>
+#include <future>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <vector>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+
+using TransactionSelector = google::spanner::v1::TransactionSelector;
+
+class KeySet {};
+class ResultSet {};
+
+// A fake `google::cloud::spanner::Client`, supporting a single `Read()`
+// operation that does nothing but track the expected transaction callbacks.
+class Client {
```

Comment:
To avoid ODR conflicts w/ the *real* `google::cloud::spanner::Client` class, what do you think about putting this one (and KeySet and ResultSet) in an unnamed namespace.

Reply:
Done.  Thanks.